### ISSUE TITLE
System.InvalidOperationException: Failed to locate opened application window with appId: PackageFamilyName!App, and processId: xxxxx.

### ISSUE THREAD

**Issue by yihsuan-chen-s** [Microsoft/WinAppDriver#606]
Hello, I'm facing a problem that if WinAppDriver.exe are executed via Jenkins.

WinAppDriver session creation:
```
    DesiredCapabilities appCapabilities = new DesiredCapabilities();
    appCapabilities.SetCapability("app", KKBOXAppId);
    appCapabilities.SetCapability("deviceName", "WindowsPC");
    Session = new WindowsDriver<WindowsElement>(new Uri("http://127.0.0.1:4723"), appCapabilities);
    Assert.IsNotNull(Session);
    Session.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(3));
```

vstest.console cmd:
```
> vstest.console.exe C:\Users\whoami\Workspace\APP\APP.UWP.UITests\bin\Debug\APP.UWP.UITests.dll /TestCaseFilter:"ClassName=APP.UWP.UITests.Test.AcceptanceTest"
```

WinAppDriver startup cmd:
```
> Start-Process 'cmd' -ArgumentList '/c "C:\Program Files (x86)\Windows Application Driver\WinAppDriver.exe" 1>WinAppDriver_Out.log 2>WinAppDriver_Err.log' -WindowStyle Hidden
```

If I execute WinAppDriver.exe via *PC's powershell* with WinAppDriver startup cmd and run test case by vstest.console cmd,
**it works perfectly.**

But if I execute WinAppDriver.exe via **Jenkins** with WinAppDriver startup cmd,
**it doesn't work.**

The app **will be started up**, but can't run any test case.
WinAppDriver got these error:
```
Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.

==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Connection: Keep-Alive
Content-Length: 116
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
{"desiredCapabilities":{"app":"PackageFamilyName!App","deviceName":"WindowsPC","platformName":"Windows"}}
HTTP/1.1 500 Internal Error
Content-Length: 168
Content-Type: application/json

==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Content-Length: 116
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
{"desiredCapabilities":{"app":"PackageFamilyName!App","deviceName":"WindowsPC","platformName":"Windows"}}
HTTP/1.1 500 Internal Error
Content-Length: 168
Content-Type: application/json
{"status":13,"value":{"error":"unknown error","message":"Failed to locate opened application window with appId: PackageFamilyName!App, and processId: 9396"}}

==========================================
```

I've tried many methods on Github's issues, such as [attaching-to-an-existing-app-window](https://github.com/Microsoft/WinAppDriver/tree/v1.0#attaching-to-an-existing-app-window), [#283](https://github.com/Microsoft/WinAppDriver/issues/283), [#213](https://github.com/Microsoft/WinAppDriver/issues/213#issuecomment-322960439), [#360](https://github.com/Microsoft/WinAppDriver/issues/360).
But all of these methods don't work for me.

Can anyone help me to resolve this issue?
Thanks

**Comment by hassanuz** [comment]
Hi @yihsuan-chen-s,

How is your Jenkin's agent setup? For UI automation to work, the agent needs to be setup as an interactive service.  

[We have guidance on setting up an agent](https://github.com/Microsoft/WinAppDriver/wiki/Continuous-Integration-with-WinAppDriver/5bee38584ade51e0d71ada221a2e9eff707a423b), but for Azure DevOps.

**Comment by yihsuan-chen-s** [comment]
Hi @hassanuz,

Thank you for your reply.
We found out that the root cause is setting of Jenkins agent services.
Our Jenkins agent is startup by Java Web Start, if we start with command line
```
$ java -jar agent.jar -jnlpUrl http://ci.internal.com/computer/windows/slave-agent.jnlp
```
WinAppDriver will work perfectly.

But if we set the Java Web Start process as a Windows Services[ (Installing Jenkins as a Windows service)](https://wiki.jenkins.io/display/JENKINS/Installing+Jenkins+as+a+Windows+service), WinAppDriver will come up with this issue `The app will be started up, but can't run any test case.`

For now, we can easily modify our setting of Jenkins agent, changing Windows Services to command lin, and package java command line as a startup script, as well as Windows Services.

Thanks again!
Regards
